crates/matrix-sdk-crypto/src/identities/user.rs

```diff
@@ -371,6 +375,19 @@ pub struct ReadOnlyUserIdentity {
     self_signing_key: SelfSigningPubkey,
 }
 
+/// Specific equality check for identities that also check if there are changes
+/// in signatures on the master cross signing keys. The default implementation
+/// of PartialEq for MasterPubKey ignores signatures (needed when importing from
+/// 4S)
```

Comment:
We should put this doc comment on the `eq()` method, this gives us the chance to inspect the `eq()` docs using the LSP:

![image](https://github.com/matrix-org/matrix-rust-sdk/assets/552026/e121917c-896e-4d61-9800-8fb35b70e98a)

It could also handle some rewording:

```rust
/// The `PartialEq` implementation compares several attributes, including the
/// user ID, key material, usage, and, notably, the signatures of the master
/// key.
///
/// This approach contrasts with the `PartialEq` implementation of the
/// [`MasterPubkey`], [`SelfSigningPubkey`] and [`UserSigningPubkey`] types,
/// where the signatures are disregarded. This distinction arises from our
/// treatment of identity as the combined representation of cross-signing keys
/// and the associated verification state.
///
/// The verification state of an identity depends on the signatures of the
/// master key, requiring their inclusion in our `PartialEq` implementation.
```

Reply:
Yes Done.
I Modified a bit the copy between Own and Other as the UserSigningKey is not a thing for others